src/main/scala/com/thirdship/libunit/AStarSolver.scala

```diff
@@ -134,103 +136,86 @@ case class AStarSolver(var allTSUnits: List[String], var allConversions: List[Co
+    *   the searched map. The openSet is iterated over until the goal is found, and populated by neighbors of units in
+    *   the openSet.
     *
-    * cameFrom is a map of units to units, storing the value of the unit was visited right before the unit key.
-    * This is used to reconstruct the path taken by the algorithm.
+    * cameFrom is a map of units to units, storing the value of the unit was visited right before the unit key in the
+    *   final path. This is used to reconstruct the path taken by the algorithm. As different ways of traveling from one
+    *   unit to another are found, the path that has the lowest cost is kept, and the higher-cost path is thrown out.
     *
     * @param start  The start unit for the search, the one being converted from.
     * @param end   The goal unit for the search, the one being converted to.
     * @return A conversion from start to goal that is the most conversion-cost-efficient.
     */
   def solve(start: String, end: String): ConversionEdge[String, Double, Double] = {
-    //println("A* algorithm begun! " + start + " is the start and " + goal + " is the goal.")
-    if(!allTSUnits.contains(start)){
-      //TODO replace with exceptions or None
-      val startNotUnit = new ScalarConversionEdge("start is","not a unit!",1)
+    if (! allTSUnits.contains(start)) {
+      val startNotUnit = new ScalarConversionEdge("start is", "not a unit!", 1) // TODO replace with exceptions or None
       return startNotUnit
     }
-    if(!allTSUnits.contains(end)){
-      //TODO replace with exceptions or None
-      val endNotUnit = new ScalarConversionEdge("end is","not a unit!",1)
+    if (! allTSUnits.contains(end)) {
+      val endNotUnit = new ScalarConversionEdge("end is", "not a unit!", 1) // TODO replace with exceptions or None
       return endNotUnit
     }
     var closedSet = List.empty[String]
     var openSet = List.empty[String]
     openSet :+= start
-    var cameFrom = Map.empty[String,String]
-    // All the units populate cameFrom as keys, and are adjusted as the algorithm finds paths.
-    var current: String = null
+    val cameFrom = mutable.Map.empty[String, String]
+    var current: String = ""
 
-    var gScore = Map.empty[String,Double]
-    var fScore = Map.empty[String,Double]
+    val gScore = mutable.Map.empty[String, Double]
+    val fScore = mutable.Map.empty[String, Double]
     gScore += start -> 0
-    fScore += start -> heuristicPM(start,end)
+    fScore += start -> heuristicPM(start, end)
 
-    while(openSet.nonEmpty){
-      openSet.sortWith(fScore.getOrElse(_,Double.PositiveInfinity) < fScore.getOrElse(_,Double.PositiveInfinity))
+    while(openSet.nonEmpty) {
+      openSet.sortWith(fScore.getOrElse(_, Double.PositiveInfinity) < fScore.getOrElse(_, Double.PositiveInfinity))
       current = openSet.head
-      // current is set to be the unit in the openSet with the lowest f score.
```

Comment:
I'll re-add them as soon as I update the ScalaStyle sheet. I removed them to save some space, but in this method especially, we should keep the comments for clarity's sake.